src/Disks/IO/CachedReadBufferFromRemoteFS.cpp

```diff
@@ -691,6 +694,34 @@ bool CachedReadBufferFromRemoteFS::updateImplementationBufferIfNeeded()
     return true;
 }
 
+bool CachedReadBufferFromRemoteFS::writeCache(char * data, size_t size, size_t offset, FileSegment & file_segment)
+{
+    Stopwatch watch(CLOCK_MONOTONIC);
+
+    try
+    {
+        file_segment.write(data, size, offset);
+    }
+    catch (ErrnoException & e)
+    {
+        int code = e.getErrno();
+        if (code == /* No space left on device */28 || code == /* Quota exceeded */122)
```

Comment:
Perhaps a little bit more idiomatic: `if (code == /* No space left on device */ ENOSPC || code == /* Quota exceeded */ EDQUOT)`

EDIT: On Darwin, macro `EDQUOT` isn't available ... using numerical codes is okay then.